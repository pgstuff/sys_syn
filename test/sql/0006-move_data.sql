BEGIN;

CREATE EXTENSION tinyint
    SCHEMA public;

CREATE EXTENSION sys_syn;

CREATE SCHEMA user_data
    AUTHORIZATION postgres;

CREATE TABLE user_data.test_table (
        test_table_id integer NOT NULL,
        test_table_text text,
        CONSTRAINT test_table_pid PRIMARY KEY (test_table_id));

INSERT INTO sys_syn.in_groups_def VALUES ('in');

SELECT sys_syn.in_table_add_sql('user_data.test_table'::regclass, 'in');

DO $$BEGIN
    EXECUTE sys_syn.in_table_add_sql('user_data.test_table'::regclass, 'in');
END$$;

INSERT INTO user_data.test_table(
        test_table_id, test_table_text)
VALUES (1,              'test_data');

INSERT INTO sys_syn.out_groups_def VALUES ('out');

SELECT sys_syn.out_table_add('user_data', 'test_table', 'out');

SELECT user_data.test_table_pull(FALSE);
SELECT user_data.test_table_out_move();

SELECT id, delta_type, queue_state FROM user_data.test_table_out_queue;

SELECT user_data.test_table_vacuum();

ROLLBACK;
