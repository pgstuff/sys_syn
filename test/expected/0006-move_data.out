BEGIN;
CREATE EXTENSION tinyint
    SCHEMA public;
CREATE EXTENSION sys_syn;
CREATE SCHEMA user_data
    AUTHORIZATION postgres;
CREATE TABLE user_data.test_table (
        test_table_id integer NOT NULL,
        test_table_text text,
        CONSTRAINT test_table_pid PRIMARY KEY (test_table_id));
INSERT INTO sys_syn.in_groups_def VALUES ('in');
SELECT sys_syn.in_table_create_sql('user_data.test_table'::regclass, 'in');
                                          in_table_create_sql                                          
-------------------------------------------------------------------------------------------------------
 SELECT sys_syn.in_table_create(                                                                      +
                 schema          => 'user_data'::regnamespace,                                        +
                 in_table_id     => 'test_table',                                                     +
                 in_group_id     => 'in',                                                             +
                 in_pull_id      => NULL,                                                             +
                 in_columns      => ARRAY[                                                            +
                        $COL$("test_table_id","integer",Id,"in_source.test_table_id",,,,)$COL$,       +
                        $COL$("test_table_text","text",Attribute,"in_source.test_table_text",,,,)$COL$+
                 ]::sys_syn.create_in_column[],                                                       +
                 full_table_reference    => 'user_data.test_table',                                   +
                 changes_table_reference => NULL,                                                     +
                 full_sql                => NULL,                                                     +
                 changes_sql             => NULL,                                                     +
                 full_pre_sql            => NULL,                                                     +
                 changes_pre_sql         => NULL,                                                     +
                 full_post_sql           => NULL,                                                     +
                 changes_post_sql        => NULL,                                                     +
                 enable_deletes_implied  => TRUE,                                                     +
                 full_prepull_id         => NULL,                                                     +
                 changes_prepull_id      => NULL                                                      +
         );
(1 row)

DO $$BEGIN
    EXECUTE sys_syn.in_table_create_sql('user_data.test_table'::regclass, 'in');
END$$;
INSERT INTO user_data.test_table(
        test_table_id, test_table_text)
VALUES (1,              'test_data');
INSERT INTO sys_syn.out_groups_def VALUES ('out');
SELECT sys_syn.out_table_create('user_data', 'test_table', 'out');
 out_table_create 
------------------
 
(1 row)

SELECT user_data.test_table_pull(FALSE);
 test_table_pull 
-----------------
 t
(1 row)

SELECT user_data.test_table_out_move();
 test_table_out_move 
---------------------
 t
(1 row)

SELECT id, delta_type, queue_state FROM user_data.test_table_out_queue;
 id  | delta_type | queue_state 
-----+------------+-------------
 (1) | Add        | Unclaimed
(1 row)

SELECT user_data.test_table_vacuum();
 test_table_vacuum 
-------------------
 
(1 row)

ROLLBACK;
